app/helpers/application_helper.rb

```diff
@@ -2,4 +2,43 @@
 
 # Any needed Application UI functions
 module ApplicationHelper
+  # Display buttons for the show page
+  # Only buttons that the user has permission to use are shown.
+  def buttons_for_show(resource)
+    buttons = []
+    buttons << edit_link(resource)
+    buttons << delete_link(resource)
+    buttons << index_link(resource)
+    safe_join(buttons.compact_blank, "|")
+  end
+
+  # Display links on Index
+  def buttons_for_index(resource)
+    buttons = []
+    buttons << edit_link(resource)
+    buttons << delete_link(resource)
+    buttons.map do |button|
+      tag.td { button }
+    end
+    safe_join(buttons)
+  end
+
+  # Delete Link
+  def delete_link(resource)
+    return unless policy(resource).destroy?
+    link_to("Delete",
+            url_for(resource),
+            method: :delete,
+            data: { confirm: "Are you sure?" })
+  end
+
+  # Edit Link
+  def edit_link(resource)
+    link_to("Edit", url_for([:edit, resource])) if policy(resource).edit?
```

Comment:
In `delete_link` you are doing an early-return guard clause, but here you are doing an inline `if` instead. Worth changing to having early-return guard clauses in `edit_link` and `index_link` to keep consistency in the helpers?